### DISCUSSION TITLE
Porespy

### DISCUSSION THREAD

**Discussion by mesbah-git** [PMEAL/porespy#693] [Q&A] [answered]
Hi there!
I have some micromodel images; I want to use your Porespy algorithm to calculate tortuosity and porosity. When I wanna use ps.simulations the JupyterHub or Colab doesn't recognize the "**simulation**" method. 
I also do dir(ps) to see its methods, and they are as follows (couldn't find simulation): 

['_Settings', '__builtins__', '__cached__', '__doc__', '__file__',  '__loader__',  '__name__',  '__package__', '__path__', '__spec__',  '__version__', '_np', 'dns', 'filters', 'generators', 'imshow',  'io', 'metrics', 'networks', 'settings', 'tools', 'visualization']

Cheers, 
Mohammad

PS. I am using Porespy 2.1.

**Comment by jgostick** [comment, marked as answer]
Hi @mesbah5959.  If you're not seeing 'simulations', then you might have an older version.  In this case you can use 'porespy.dns.tortuosity', which is now deprecated in favor of 'porespy.simulations.tortuosity_fd', but it still works.
